src/Hedgehog/Gen.fs

```diff
@@ -447,6 +447,267 @@ module Gen =
+    let cSeq (lower : int) (upper : int) : (Gen<'a> -> Gen<seq<'a>>) =
+        seq (Range.constant lower upper)
+
+    /// Shortcut for Gen.string (Range.exponential lower upper).
+    let eString (lower : int) (upper : int) : (Gen<char> -> Gen<string>) =
+        string (Range.exponential lower upper)
+
+    /// Shortcut for Gen.string (Range.linear lower upper).
+    let lString (lower : int) (upper : int) : (Gen<char> -> Gen<string>) =
+        string (Range.linear lower upper)
+
+    /// Shortcut for Gen.string (Range.constant lower upper).
+    let cString (lower : int) (upper : int) : (Gen<char> -> Gen<string>) =
+        string (Range.constant lower upper)
+
+    /// Generates null part of the time.
+    let withNull (g : Gen<'a>) : Gen<'a> =
+        g |> option |> map (fun xOpt ->
+        match xOpt with Some x -> x | None -> null)
+
+    /// Generates a value that is not null.
+    let noNull (g : Gen<'a>) : Gen<'a> =
+        g |> filter (not << isNull)
+
+    /// Generates a value that is not equal to another value.
+    let notEqualTo (other : 'a) : (Gen<'a> -> Gen<'a>) =
+        filter ((<>) other)
+
+    /// Generates a value that is not equal to another option-wrapped value.
+    let notEqualToOpt (other : 'a option) : (Gen<'a> -> Gen<'a>) =
+        filter (fun x -> match other with Some o -> x <> o | None -> true)
+
+    /// Generates a value that is not contained in the specified list.
+    let notIn (list: 'a list) (g : Gen<'a>) : Gen<'a> =
+        g |> filter (fun x -> not <| List.contains x list)
+
+    /// Generates a list that does not contain the specified element.
+    /// Shortcut for Gen.filter (not << List.contains x)
+    let notContains (x: 'a) : (Gen<'a list> -> Gen<'a list>) =
+      filter (not << List.contains x)
+
+    /// Generates a string that is not equal to another string using
+    /// StringComparison.OrdinalIgnoreCase.
+    let iNotEqualTo (str : string) : (Gen<string> -> Gen<string>) =
+        filter <| fun s ->
+            not <| str.Equals(s, System.StringComparison.OrdinalIgnoreCase)
+
+    /// Generates a string that is not a substring of another string.
+    let notSubstringOf (str : string) : (Gen<string> -> Gen<string>) =
+      filter <| fun s -> not <| str.Contains s
+
+    /// Generates a string that is not a substring of another string using
+    /// StringComparison.OrdinalIgnoreCase.
+    let iNotSubstringOf (str : string) : (Gen<string> -> Gen<string>) =
+      filter <| fun s ->
+          str.IndexOf(s, System.StringComparison.OrdinalIgnoreCase) = -1
+
+    /// Generates a string that does not start with another string.
+    let notStartsWith (str : string) : (Gen<string> -> Gen<string>) =
+      filter <| fun s -> not <| s.StartsWith str
+
+    /// Generates a string that does not start with another string using
+    /// StringComparison.OrdinalIgnoreCase.
+    let iNotStartsWith (str : string) : (Gen<string> -> Gen<string>) =
```

Comment:
Perhaps `iNotStartsWith` a bit cryptic 😕 Could we come up with a more descriptive name? 😕